libxbrzscale.cpp

```diff
@@ -180,29 +206,48 @@ void libxbrzscale::uint32toSurface(uint32_t* ui32src, SDL_Surface* dst_img){
 }
 
 SDL_Surface* libxbrzscale::scale(SDL_Surface* src_img, int scale){
+  return libxbrzscale::scale(NULL, src_img, scale);
+}
+/**
+ * dst_img if not null may be re-used, and is also returned
+ */
+SDL_Surface* libxbrzscale::scale(SDL_Surface* dst_imgCache, SDL_Surface* src_img, int scale){
+  if(scale<2 || scale>6){std::cerr<<"invalid stretch value min=2, max=6, requested="<<scale<<std::endl;exit(1);}
```

Comment:
Please don't use console output.
Since it's a class now, better way to return an error would be to have an errorCode and maybe errorMessage property that you are supposed to check if this function returns NULL for example.
exit(1) from a library because some parameter was bad is not common practice and I would have a WTF moment if something like this happend to me.